src/Playwright/Core/Waiter.cs

```diff
@@ -122,6 +122,15 @@ internal void RejectImmediately(Exception exception)
         string e,
         PlaywrightException navigationException,
         Func<T, bool> predicate = null)
+    {
+        RejectOnEvent(eventSource, e, () => navigationException, predicate);
+    }
+
+    internal void RejectOnEvent<T>(
+        object eventSource,
+        string e,
+        Func<PlaywrightException> navigationExceptionPredicate,
```

Comment:
Why can we pass `navigationException` directly instead of a function? Also predicate sounds like a wrong term for this parameter.